test/accessor_tests.cc

```diff
@@ -641,5 +656,38 @@ namespace detail {
 		sycl::free(result, q.get_sycl_queue());
 	}
 
+	TEST_CASE_METHOD(test_utils::runtime_fixture, "out of bounds access gets reported.", "[accessor][oob]") {
+#if !CELERITY_ACCESSOR_BOUNDARY_CHECK
+		SKIP("accessor boundary check is disabled.");
+#endif
+		std::vector<int> original_values{1, 2};
+		std::vector<int> validation_values(2);
+		std::vector<int> zeros{0, 0};
+		buffer<int, 1> buff_0(original_values.data(), range<1>{2});
+		buffer<int, 1> buff_1(zeros.data(), range<1>{2});
+		distr_queue q;
+
+		q.submit([&](handler& cgh) {
+			accessor acc0(buff_0, cgh, celerity::access::one_to_one(), celerity::read_only);
+			accessor acc1(buff_1, cgh, celerity::access::one_to_one(), celerity::write_only, celerity::no_init);
+			cgh.parallel_for<class UKN(acc_out_of_bounds)>(buff_0.get_range(), [=](item<1> it) {
+				const auto odd = it.get_id() + 1;
+				acc1[it] = acc0[odd];
+			});
+		});
+		q.slow_full_sync();
+
+		q.submit([&](handler& cgh) {
+			accessor acc1(buff_1, cgh, celerity::access::all(), celerity::read_only_host_task);
+			cgh.host_task(on_master_node, [=, &validation_values] {
+				validation_values[0] = acc1[0];
+				validation_values[1] = acc1[1];
+			});
+		});
+		q.slow_full_sync();
+		CHECK(validation_values[1] == original_values[0]);
```

Comment:
As far as I can tell, this only works because of the current behavior of OOB-accesses returning a reference to `ptr[0]`. That is pretty arbitrary and IMO we shouldn't be testing it. Instead I think this test should explicitly check for the error message being emitted using a `log_capture`!